### PR TITLE
Add more `instances.go` unit tests

### PR DESCRIPTION
**Pull request by mihaelabalutoiu**
Add more `instances.go` unit tests

Signed-off-by: Mihaela Balutoiu <mbalutoiu@cloudbasesolutions.com>

### CODE REVIEW

File: database/sql/instances_test.go [hunk 1]
```diff
@@ -73,10 +86,12 @@ func (s *InstancesTestSuite) SetupTest() {
 			context.Background(),
 			pool.ID,
 			params.CreateInstanceParams{
-				Name:        fmt.Sprintf("test-instance-%v", i),
-				OSType:      "linux",
-				OSArch:      "amd64",
-				CallbackURL: "https://garm.example.com/",
+				Name:         fmt.Sprintf("test-instance-%v", i),
```
Comment (gabriel-samfira) [inline database/sql/instances_test.go:89]: I think you can replace ```%v``` with ```%d```

File: database/sql/instances_test.go [hunk 2]
```diff
@@ -164,6 +179,115 @@ func (s *InstancesTestSuite) TestGetInstanceByNameFetchInstanceFailed() {
...
+	storeInstance := s.Fixtures.Instances[0]
+	err := s.Store.DeleteInstance(context.Background(), s.Fixtures.Pool.ID, storeInstance.Name)
+
+	s.Require().Nil(err)
+
+	_, err = s.Store.GetPoolInstanceByName(context.Background(), s.Fixtures.Pool.ID, s.Fixtures.Instances[0].Name)
+	s.Require().Equal("fetching instance: fetching pool instance by name: not found", err.Error())
+}
+
+func (s *InstancesTestSuite) TestDeleteInstanceFailed() {
```
Comment (gabriel-samfira) [inline database/sql/instances_test.go:192]: This mostly tests if ```DeleteInstance()``` validates the pool ID. Should be renamed to something like: ```TestDeleteInstanceInvalidPoolID```

Same with a few more functions bellow.

Comment (gabriel-samfira) [reply database/sql/instances_test.go:192]: If you want to test a condition where the database returns an error, you have two options:

  * Use mocks
  * Make/open the database read-only and try to mutate the data

**Review by gabriel-samfira — CHANGES_REQUESTED** [review]

File: database/sql/instances_test.go (hunk 2 above)
Comment (mihaelabalutoiu) [reply database/sql/instances_test.go]: > This mostly tests if `DeleteInstance()` validates the pool ID. Should be renamed to something like: `TestDeleteInstanceInvalidPoolID`
> 
> Same with a few more functions bellow.

Fixed

Comment (mihaelabalutoiu) [reply database/sql/instances_test.go]: > If you want to test a condition where the database returns an error, you have two options:
> 
> * Use mocks
> * Make/open the database read-only and try to mutate the data

Sounds good. I will add such a test case in a separate PR. Thank-you for the suggestion.

**Pull request merged**